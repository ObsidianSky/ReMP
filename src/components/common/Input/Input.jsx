import React from 'react';
import PropTypes from 'prop-types';
import './Input.scss'

export default class Input extends React.Component {
    render() {
        return (
            <input type="text" className="input" value="some text"/>
        )
    }
}

Input.propTypes = {};