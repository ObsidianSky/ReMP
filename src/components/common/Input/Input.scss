@import '../../../assets/styles/variables';

.input {
  display: block;
  width: 100%;
  border: none;
  font-size: 18px;
  background: $almost-black;
  color: $text;
  border-bottom: 2px solid $pink;
  padding: 13px 20px;
}