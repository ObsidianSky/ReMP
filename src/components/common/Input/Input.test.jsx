import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Input from './Input';

describe('Input', () => {
    let props;
    let onChange;

    beforeEach(() => {
        onChange = jest.fn();

        props = {
            value: 'value',
            placeholder: 'placeholder',
            onChange
        }
    });

    test('should renders correctly', () => {
        const wrapper = shallow(<Input { ...props }/>);

        expect(toJson(wrapper)).toMatchSnapshot();
    });

    test('should call on change callback with input value', () => {
        const inputValue = 'input value';
        const wrapper = shallow(<Input { ...props }/>);

        wrapper.find('input').simulate('change', { target: { value: inputValue } });

        expect(onChange).toHaveBeenCalledWith(inputValue)
    });
});
