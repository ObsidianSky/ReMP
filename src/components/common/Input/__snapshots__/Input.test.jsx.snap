// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should renders correctly 1`] = `
<input
  className="input"
  onChange={[Function]}
  placeholder="placeholder"
  type="text"
  value="value"
/>
`;
